examples/TransactionGenerator.hs

```diff
@@ -299,92 +555,194 @@ mainInfo =
-    =<< runClientM (poll (Poll (_rkRequestKeys rkeys))) (_genClientEnv gencfg)
-pollRequestKeys _ _ = error "Need exactly one request key"
+      (runReaderT (runTransactionGenerator (simpleloop measure)) gencfg)
+      (TransactionGeneratorState gen 0)
+
+pollRequestKeys :: MeasureTime -> RequestKeys -> ScriptConfig -> IO ()
+pollRequestKeys (MeasureTime mtime) rkeys@(RequestKeys [_]) config = do
+  (timeTaken, !_action) <- measureDiffTime go
+  when mtime (putStrLn $ "" <> show timeTaken)
+  where
+    go = do
+      putStrLn "Polling your requestKey"
+      gencfg <- mkTransactionGeneratorConfig Nothing config
+      flip runReaderT gencfg $ do
+        poll <- views genServantRecord apiPoll
+        clientEnv <- view genClientEnv
+        response <-
+          liftIO $ runClientM (poll (Poll (_rkRequestKeys rkeys))) clientEnv
+        liftIO $ case response of
+          Left _ -> do
+            putStrLn "Failure"
+            exitWith (ExitFailure 1)
+          Right (PollResponses a) ->
+            if null a
+              then do
+                putStrLn "Failure no result returned"
+                exitWith (ExitFailure 1)
+              else do
+                print a
+                exitSuccess
+pollRequestKeys _ _ _ = error "Need exactly one request key"
+
+listenerRequestKey :: MeasureTime -> ListenerRequest -> ScriptConfig -> IO ()
+listenerRequestKey (MeasureTime mtime) listenerRequest config = do
+    (timeTaken, response) <- measureDiffTime go
+    when mtime (putStrLn $ "" <> show timeTaken)
+    case response of
+        Left err -> do
+          print err
+          exitWith (ExitFailure 1)
+        Right r -> do
+          print (_arResult r)
+          exitSuccess
+  where
+    go = do
+      putStrLn "Listening..."
+      gencfg <- mkTransactionGeneratorConfig Nothing config
+      let listen = apiListen . _genServantRecord $ gencfg
+      runClientM (listen listenerRequest) (_genClientEnv gencfg)
 
 main :: IO ()
 main =
   runWithConfiguration mainInfo $ \config ->
-    case _scriptCommand config of
-      NoOp -> putStrLn "NoOp: You probably don't want to be here."
-      DeployContracts contracts ->
-        loadContracts (initAdminKeysetContract : ((createLoader <$> contracts) `go` defaultContractLoaders)) config
-        where
-          go xs ys = if null xs then ys else xs
-      RunStandardContracts -> sendTransactions config
-      RunSimpleExpressions -> sendSimpleExpressions config
-      PollRequestKeys requestKeys -> pollRequestKeys (RequestKeys (map (RequestKey . Hash) requestKeys)) config
-
--- TOOD: This is here for when a user wishes to deploy their own
+    if HS.member (_nodeChainId config) $ chainIds_ $ _chainGraph (_nodeVersion config)
+       then startup config
+       else error $ "This chain: " <> show (_nodeChainId config) <> " is invalid given this chainweb version: " <> show (_nodeVersion config)
+  where
+    startup config = do
+      let transHandle = config ^. logHandleConfig
```

Comment:
Don't worry about changing it now, but for future reference...in general if you're only reading, then the accessor will be more efficient and optimizable by the compiler, and easier to understand than lens operator soup.  In other words, don't use a lens when a function call will do.

Reply:
Already changed.